crates/project/src/project_tests.rs

```diff
@@ -4909,6 +4916,120 @@ async fn test_multiple_language_server_actions(cx: &mut gpui::TestAppContext) {
     );
 }
 
+#[gpui::test]
+async fn test_reordering_worktrees(cx: &mut gpui::TestAppContext) {
+    init_test(cx);
+
+    let fs = FakeFs::new(cx.executor());
+    fs.insert_tree(
+        "/dir",
+        json!({
+            "a.rs": "let a = 1;",
+            "b.rs": "let b = 2;"
+        }),
+    )
+    .await;
+
+    let project = Project::test(fs, ["/dir/a.rs".as_ref(), "/dir/b.rs".as_ref()], cx).await;
```

Comment:
Not insisting, but as an idea: having 3 items might cover more states and unveil more n+1 issues in general.
Here it seems fine now, but if you remember the old code with the bug that I've filmed, this test might not cover all issues with it.

Reply:
Sounds good, I'll update the test to cover that!